nncf/quantization/algorithms/accuracy_control/ranker.py

```diff
@@ -0,0 +1,314 @@
+        """
+        :param ranking_subset_size: The number of data items that will be selected from
+            the dataset to rank groups of quantizers. The `len(dataset)` data items will
+            be selected if `ranking_subset_size` parameter is greater than the number of
+            elements in the dataset.
+        :param ranking_fn: A function that compares values returned by
+            `_collect_values_for_each_item()` for initial and quantized models.
+        :param dataset: Dataset for the ranking process.
+        :param algo_backend: The `AccuracyControlAlgoBackend` algo backend.
+        """
+        self._ranking_subset_size = ranking_subset_size
+        self._ranking_fn = ranking_fn
+        self._dataset = dataset
+        self._algo_backend = algo_backend
+        # We don't need to re-calculate values for the initial model
+        # because they don't change. So use this attribute to store
+        # them to improve execution time.
+        self._ref_values = None
+
+    def find_groups_of_quantizers_to_rank(self, quantized_model_graph: NNCFGraph) -> List[GroupToRank]:
+        """
+        Finds groups of quantizers to rank.
+
+        :param quantized_model_graph: Graph for quantized model.
+        :return: List of groups of quantizers to rank.
+        """
+        groups_to_rank = []
+        processed = {}
+        # TODO(andrey-churkin): Set order of quantizers here.
+        quantizers = quantized_model_graph.get_nodes_by_metatypes(self._algo_backend.get_quantizer_metatypes())
+        for quantizer_node in quantizers:
+            if processed.get(quantizer_node.node_name, False):
+                continue
+            quantizers, operations = find_quantizer_nodes_to_cut(quantized_model_graph,
+                                                                 quantizer_node,
+                                                                 self._algo_backend.get_quantizer_metatypes(),
+                                                                 self._algo_backend.get_const_metatypes(),
+                                                                 self._algo_backend.get_quantizable_metatypes(),
+                                                                 self._algo_backend.get_quantize_agnostic_metatypes(),
+                                                                 self._algo_backend.get_shape_of_metatypes())
+            for x in quantizers:
+                processed[x.node_name] = True
+
+            groups_to_rank.append(GroupToRank(quantizers, operations))
+
+        return groups_to_rank
+
+    def rank_groups_of_quantizers(self,
+                                  groups_to_rank: List[GroupToRank],
+                                  initial_model: TModel,
+                                  quantized_model: TModel,
+                                  quantized_model_graph: NNCFGraph,
+                                  excluded_groups: Optional[GroupToRank] = None) -> List[GroupToRank]:
```

Comment:
Could you briefly explain why you use `excluded_groups` instead of updating `groups_to_rank`?